python-packages/presto_drf/presto_drf/tests/test_parsers.py

```diff
@@ -0,0 +1,92 @@
+import io
+
+from django.core.files.uploadhandler import MemoryFileUploadHandler
+from django.test import SimpleTestCase
+
+from presto_drf.parsers import CamelCaseJSONParser
+from presto_drf.parsers import CamelCaseMultiPartJSONParser
+
+
+class TestParsers(SimpleTestCase):
+    def bytes(self, value):
+        return io.BytesIO(value.encode())
+
+    def test_camel_case_json_parser(self):
+        parser = CamelCaseJSONParser()
+        result = parser.parse(self.bytes('{"dataKeyFoo":{"innerKeyBar":1,"key":2}}'))
+        self.assertEqual(result, {"data_key_foo": {"inner_key_bar": 1, "key": 2}})
+
+        # Numbers in keys were problematic with old djangorestframework_camel_case.parser.CamelCaseJSONParser
+        # Test that we preserve whatever the input convetion was
+        result = parser.parse(
+            self.bytes('{"keyWithNumeric_1":{"keyWithNumeric2":1,"key":2}}')
+        )
+        self.assertEqual(
+            result, {"key_with_numeric_1": {"key_with_numeric2": 1, "key": 2}}
+        )
+
+    def test_camel_case_json_parser_underscorize_ignore(self):
```

Comment:
Minor since it's in a test case but this uses underscorize whereas everything else uses underscor*e*ize